### PR TITLE
Fix check for file reference being in use

### PR DESCRIPTION
**Pull request by hmusum**
Do this in a way that actually works.

### CODE REVIEW

File: configserver/src/main/java/com/yahoo/vespa/config/server/filedistribution/FileDirectory.java [hunk 1]
```diff
@@ -127,13 +127,13 @@ public FileReference addFile(File source) throws IOException {
...
     }
 
-    public void delete(FileReference fileReference, Function<FileReference, Boolean> canBeDeleted) {
+    public void delete(FileReference fileReference, ApplicationRepository applicationRepository) {
         if (useLock.value())
             try (Lock lock = locks.lock(fileReference)) {
-                if (canBeDeleted.apply(fileReference))
-                    deleteDirRecursively(destinationDir(fileReference));
-                else
+                if (applicationRepository.isFileReferenceInUse(fileReference))
```
Comment (baldersheim) [inline configserver/src/main/java/com/yahoo/vespa/config/server/filedistribution/FileDirectory.java:133]: It was reversed ?

**Review by baldersheim — APPROVED** [review]

**Review by baldersheim — CHANGES_REQUESTED** [review]
Requires a fix in configserver too, build not happy.

File: configserver/src/main/java/com/yahoo/vespa/config/server/filedistribution/FileDirectory.java (hunk 1 above)
Comment (hmusum) [reply configserver/src/main/java/com/yahoo/vespa/config/server/filedistribution/FileDirectory.java:133]: No, expression was evaluated immediately, not when calling `apply`.

**Comment by hmusum** [comment]
A seconds commit was missing

**Comment by hmusum** [comment]
@baldersheim please take a look again

**Pull request closed without merging**